### ISSUE TITLE
Assets not present in the generated project

### ISSUE THREAD

**Issue by igorkulman** [tuist/tuist#138]
**Describe the bug**
When using `tuist generate` the resulting project does not include the `Assets.xcassets` folder although in the `Sources` folder.

**To Reproduce**
1. Directory structure
<img width="676" alt="screen shot 2018-09-25 at 11 15 45" src="https://user-images.githubusercontent.com/4746070/46004916-5fe09480-c0b4-11e8-8b40-617b3ce12de7.png">

2. run `tuist generate`
3. Result
<img width="174" alt="screen shot 2018-09-25 at 11 16 13" src="https://user-images.githubusercontent.com/4746070/46004957-71c23780-c0b4-11e8-80a8-70b549a5b992.png">

**Expected behavior**
`Assets.xcassets`  should be included in the generated project

**Desktop (please complete the following information):**
 - OS: macOS
 - Version: 10.13.6

**Additional context**
Config file:

```
import ProjectDescription

let project = Project(name: "iOSSampleApp",
                      targets: [
                        Target(name: "iOSSampleApp",
                               platform: .iOS,
                               product: .app,
                               bundleId: "sk.kulman.iOSSampleApp",
                               infoPlist: "Info.plist",
                               sources: "Sources/**",
                               dependencies: [
                                    .framework(path: "Carthage/Build/iOS/RxSwift.framework"),
                                    .framework(path: "Carthage/Build/iOS/RxSwiftExt.framework"),
                                    .framework(path: "Carthage/Build/iOS/RxCocoa.framework"),
                                    .framework(path: "Carthage/Build/iOS/RxSwift.framework"),
                                    .framework(path: "Carthage/Build/iOS/FeedKit.framework"),
                                    .framework(path: "Carthage/Build/iOS/Reusable.framework"),
                                    .framework(path: "Carthage/Build/iOS/Defaults.framework"),
                                    .framework(path: "Carthage/Build/iOS/CleanroomLogger.framework"),
                                    .framework(path: "Carthage/Build/iOS/Swinject.framework"),
                                    .framework(path: "Carthage/Build/iOS/SwinjectAutoregistration.framework"),
                                    .framework(path: "Carthage/Build/iOS/SwinjectStoryboard.framework"),
                                    .framework(path: "Carthage/Build/iOS/CRToast.framework"),
                                    .framework(path: "Carthage/Build/iOS/Nuke.framework"),
                                    .framework(path: "Carthage/Build/iOS/RxNuke.framework")
                                ]),
                        Target(name: "iOSSampleAppTests",
                               platform: .iOS,
                               product: .unitTests,
                               bundleId: "sk.kulman.iOSSampleAppTests",
                               infoPlist: "Tests.plist",
                               sources: "Tests/**",
                               dependencies: [
                                    .target(name: "iOSSampleApp"),
                                    .framework(path: "Carthage/Build/iOS/RxTest.framework"),
                                    .framework(path: "Carthage/Build/iOS/RxBlocking.framework"),
                                    .framework(path: "Carthage/Build/iOS/Quick.framework"),
                                    .framework(path: "Carthage/Build/iOS/Nimble.framework")
                               ]),
                        Target(name: "iOSSampleAppUITests",
                               platform: .iOS,
                               product: .unitTests,
                               bundleId: "sk.kulman.iOSSampleUIAppTests",
                               infoPlist: "UITests.plist",
                               sources: "UITests/**",
                               dependencies: [
                                .target(name: "iOSSampleApp"),
                                .framework(path: "Carthage/Build/iOS/SimulatorStatusMagiciOS.framework")
                            ])
                      ])
```

**Comment by pepibumur** [comment]
The sources attribute of the target is just for code files *(e.g. .swift, .m, .mm)*. There's another attribute for resources, `resources`, which you should use instead. You can check out [the constructor](https://github.com/tuist/tuist/blob/master/Sources/TuistKit/Models/Target.swift#L28) to see arguments that you can pass.

**Comment by igorkulman** [comment]
In the referenced file I see `resources: [AbsolutePath] = [],` which is great, I need two paths, one for images and another for localized strings but with

```
resources: [
    "Sources/Assets.xcassets/**",
    "Sources/Resources/en/*"
]
```

I get

```
 Error: /var/folders/bt/ckwd2h017zd6hl4gnnfhxrv80000gn/T/TemporaryFile.7gyDO3:11:43: error: cannot convert value of type '[String]' to expected argument type 'String?'
                               resources: [
                                          ^
```

**Comment by pepibumur** [comment]
The idea behind resources being a string is creating a convention where all resources are organized under one directory. We could have been more flexible but we intentionally wanted to make developers think about the folders structure and keep it clean and well structured.

Is this a convention thar you could adopt or do you think it’s too restrictive?

**Comment by igorkulman** [comment]
I think it should be fine for most people. 

For my sample project I am trying to convert (https://github.com/igorkulman/iOSSampleApp) it means copying all the assets, strings and storyboard to one separate directory, which is not ideal. 

I like to organize my project according to "scenarios", so everything for say [setup](https://github.com/igorkulman/iOSSampleApp/tree/master/iOSSampleApp/Scenarios/Setup) (stoyboard, cells, viewcontrolles, viewmodels, etc) in one directory, instead of organizing the project by file type (all storyboard in one directory, all viewcontrollers in another, etc.).

So it would be nice to have that option.

**Comment by pepibumur** [comment]
Would something like [CocoaPods file patterns](https://guides.cocoapods.org/syntax/podspec.html#group_file_patterns) work? Looking at that example app, those "scenarios" could be expressed in the manifest as:

```swift
Target(sources: "iOSSampleApp/**/*.{swift}", 
       resources: "iOSSampleApp/**/*.{strings, png, storyboard, xib}")
```

Do you think that it helps or that it makes the definition more complex? My only concern regarding accepting an array as an input is that one can end up with a messy folders structure. By limiting the input of that attribute Tuist makes the developer reason about how assets are organized. If you think compromises the readability of the manifest, then I'd consider what you suggested.

**Comment by pepibumur** [comment]
What do you think about [SPM's approach](https://github.com/apple/swift-package-manager/blob/master/Sources/PackageDescription4/Target.swift#L87)? It accepts the following attributes:

- Path: `String?` Path to the directory where the files are.
- Included: `[String]?` Glob patterns relative from the path attribute to be included.
- Excluded: `[String]?` Glob patterns relative from the path attribute to be excluded.

Tuist's target could be like:

```swift
Target(path: "iOSSampleApp",
  sources: [.included("**/*.swift")],
  resources: [.included("Resources/**/*"), .excluded("*.plist")]
)
```

**Comment by igorkulman** [comment]
This format seems reasonable, but will the generated Code project split the files between `Sources` and `Resources`. So if I have say `UserCell.swift` and `UserCell.xib` in `Scenarios\Dashboard\Cells` will I find those files next to each other in the Xcode project? Or will `UserCell.swift` be in `Sources` and `UserCell.xib` in `Resources`? If yes then it is no improvement for me personally, because it will keep my preferred file structure only on the disk and not in Xcode.

**Comment by igorkulman** [comment]
Sounds good

**Comment by ladislas** [comment]
I'm not sure I understand everything that's being said, but IMO it's important to be able to customize where your resources are.

In our case, our app uses a lot of external content, videos, images, yaml/json files, etc., that we keep in different submodules. The goal is to make them independent, easily modifiable without modifying the main app repository (besides updating the submodules). It also helps us keeping the main source repository 'light'.

I guess I could put all the different submodules inside a big `Resources` directory but that would be a big change.

The SPM approach your mentioned where you can specify and array of resources with different paths is a great approach as you can cherry pick what you need/want.

**Comment by pepibumur** [comment]
Thanks for bringing up your concerns @ladislas. Answering/commenting some of the things that you mentioned.

> In our case, our app uses a lot of external content, videos, images, yaml/json files, etc., that we keep in different submodules. The goal is to make them independent, easily modifiable without modifying the main app repository (besides updating the submodules). It also helps us keeping the main source repository 'light'.

If you are using submodules, you'll find very useful the new action that we plan to add `tuist up`. When executed, it'll set up your project to start working on it. Each project could define how to be set up, which in your case would be pulling the git submodules.

> I guess I could put all the different submodules inside a big Resources directory but that would be a big change. The SPM approach your mentioned where you can specify and array of resources with different paths is a great approach as you can cherry pick what you need/want.

With that API we aim to make easier to select which files should be included/excluded. It does not allow to customize how those files will be structured in the Xcode project. The convention that we'll follow in groups is that files will be before folders and both will be sorted alphabetically. The structure will match the folders structure in your system to avoid the problem Xcode projects historically had with a messy groups structure that didn't match the folders in the file system.

**Comment by ollieatkinson** [comment]
Fixed by #266
